### PR TITLE
Sus 179/adding billing endpoints

### PR DESCRIPTION
**Pull request by JiaanNie**
still need to create the return type

### CODE REVIEW

File: src/services/products/billing/billing.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  axiosClient: any
+  logger: Logger
+
+  constructor(axiosClient: any, logger: Logger) {
+    this.axiosClient = axiosClient;
+    this.logger = logger;
+  }
+
+  createBillingAccount(): void {
+    return this.axiosClient.post(`/products/billing/create_billing_account`).then((res: any) => {
```
Comment (ShawnClake) [inline src/services/products/billing/billing.ts]: The return types for all these functions should be a promise

**Review by ShawnClake — CHANGES_REQUESTED** [review]

File: src/services/products/billing/types/billingResponseTypes.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export interface BillingAccountResponse{
+  first_name: string,
```
Comment (ghost) [inline src/services/products/billing/types/billingResponseTypes.ts:2]: Should these be snake case or camel case here?

File: src/services/sam/supports/types/ticketReponseTypes.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export interface CreateTicketResponse {
+  ticket_id: string,
+  subject: string,
+  priority: string
+}
+export interface GetTicketResponse extends  CreateTicketResponse {}
```
Comment (ghost) [inline src/services/sam/supports/types/ticketReponseTypes.ts:6]: Spacing

File: src/utils/logger.ts [hunk 4]
```diff
@@ -14,15 +16,17 @@ export class Logger {
...
 
-    debug(message: string) {
+    debug(message: string | AxiosResponse) {
         if(this._logLevel >= LogLevel.DEBUG) {
             console.log(`[eHelply SDK] ${message}`)
         }
     }
 
-    info(message: string) {
+
```
Comment (ghost) [inline src/utils/logger.ts]: Extra space here

File: tsconfig.json [hunk 5]
```diff
@@ -14,7 +14,7 @@
     "target": "es5",                                     /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */
     // "lib": [],                                        /* Specify a set of bundled library declaration files that describe the target runtime environment. */
     // "jsx": "preserve",                                /* Specify what JSX code is generated. */
-    // "experimentalDecorators": true,                   /* Enable experimental support for TC39 stage 2 draft decorators. */
+    // "experimentalDecorators": true,                   /* Enable experimental supports for TC39 stage 2 draft decorators. */
```
Comment (ghost) [inline tsconfig.json:17]: Was this supposed to change?

Comment (JiaanNie) [reply tsconfig.json:17]: i think this was coming from when we merged support into master, then i did a rebase  master in the current branch

File: src/sdk.ts [hunk 6]
```diff
@@ -98,7 +100,8 @@ export class eHelplySDK {
     private createServices(): void {
         this.services = {
             users: new UserSdk(this._axiosClient, this._logger),
-            reviews: new ReviewSdk(this._axiosClient, this._logger)
-        };
+            reviews: new ReviewSdk(this._axiosClient, this._logger),
+            support: new SupportSdk(this._axiosClient, this._logger)
+    };
```
Comment (ShawnClake) [inline src/sdk.ts:105]: The indenting of this closing curly is incorrect

File: src/services/products/billing/billing.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  createBillingAccount(): Promise<BillingAccountResponse> {
+    return this.axiosClient.post<BillingAccountResponse>(
+      `/products/billing/create_billing_account`
+    ).then((res: AxiosResponse<BillingAccountResponse>) => {
+      this.logger.debug(res);
+      return res.data
+    });
+  }
+
+  getSecret(): Promise<GetSecretResponse> {
```
Comment (ShawnClake) [inline src/services/products/billing/billing.ts]: This name needs to be improve. Perhaps `getStripeSecret`?

File: src/services/products/billing/types/billingResponseTypes.ts (hunk 2 above)
Comment (ShawnClake) [reply src/services/products/billing/types/billingResponseTypes.ts:2]: Depends on the pydantic model. Our python standards dictate using snake_case

File: src/services/users/users.ts [hunk 8]
```diff
@@ -22,6 +22,8 @@ export default class UserSdk {
         }
         return this.axiosClientHeaderless.post(`/users/auth/login`, data).then((res: any) => {
             this.logger.debug(res);
+        }).catch(error => {
```
Comment (ShawnClake) [inline src/services/users/users.ts]: We shouldnt have a catch here. Also why are we modifying `users.ts` code in this PR? Unintentional? If so, revert these changes.

File: tsconfig.json (hunk 5 above)
Comment (ShawnClake) [reply tsconfig.json:17]: It's a comment, it probably doesnt matter

**Review by ShawnClake — CHANGES_REQUESTED** [review]
Like I said in the other PR, leverage ctrl-alt-L to auto format your files.

File: src/services/products/billing/billing.ts (hunk 7 above)
Comment (JiaanNie) [reply src/services/products/billing/billing.ts]: yeah that fair

File: src/services/users/users.ts (hunk 8 above)
Comment (JiaanNie) [reply src/services/users/users.ts]: i didnt change user.ts it couldnt of been the branch was behind and i didnt update the current branch

Comment (JiaanNie) [reply src/services/users/users.ts]: all the fixes is in the user branch of the sdk once we merge it the problem will go away

Comment (ShawnClake) [reply src/services/users/users.ts]: I think youll get a merge conflict here if you don't remove the change to this code that is on this branch. It's highlighted green because it's a change that was made on this branch.

**Review by ShawnClake — APPROVED** [review]
Merge at your own risk for a merge conflict with the users.ts stuff. Since I don't have to resolve any merge conflicts, I'm fine if you want to sort through it that way. There's some inconsistent indentation levels and spacing that still remains, and that should be fixed in the future, but now that all the names are correct, I'm willing to let that slide for now.
